routes/api/v1/forecast.js

```diff
@@ -0,0 +1,68 @@
+var express = require('express');
+var router = express.Router();
+const fetch = require('node-fetch');
+const dotenv = require('dotenv');
+dotenv.config();
+const { darkSkyApi, googleApi } = require('../../../config/config');
+pry = require('pryjs');
+
+class Forecast {
+  constructor(forecast_params){
+    this.location = forecast_params.location
+  }
+
+  weather(weather, req) {
+    let output = {
+      'location': req.query.location.toUppercase(),
+      'currently': weather.currently,
+      'hourly': weather.hourly,
+      'daily': weather.daily
+    }
+    return output
+  }
+
+  googleService() {
+    const fetchPromise = fetch(`https://maps.googleapis.com/maps/api/geocode/json?key=${googleApi}&address=${this.location}`)
+    .then(response => {return response.json();
+    })
+    .then(locationInfo => {
+      return  locationInfo.results[0].geometry.location
+    })
+    .catch(error => {
+      {error}
+    })
+    return fetchPromise
+  }
+
+  darkskyService(coords) {
+    return fetch(`https://api.darksky.net/forecast/${darkSkyApi}/${coords.lat},${coords.lng}?exclude=minutely,alerts,flags`)
+    .then( response => response.json())
+    .then( weatherInfo => {
+      return weatherInfo
+    })
+    .catch(error => {
+      {error}
+    })
+  }
+}
+
+router.get('/', function(req, res, next){
+  if (req.body.hasOwnProperty('api_key')){
```

Comment:
Is the api_key submitted with the request authenticated?

Reply:
It is not, probably not very secure ill look into refactoring.